spec/features/workflow_service_creation_spec.rb

```diff
@@ -1,25 +1,22 @@
 require 'spec_helper'
 
-RSpec.feature 'Workflow Service Creation' do
-  let(:current_user) do
-    double(
-      :webauth_user,
-      login: 'sunetid',
-      logged_in?: true,
-      permitted_apos: [],
-      is_admin: true,
-      roles: []
-    )
-  end
-  before do
-    allow_any_instance_of(ItemsController).to receive(:current_user)
-      .and_return(current_user)
-    allow_any_instance_of(CatalogController).to receive(:current_user)
-      .and_return(current_user)
+feature 'Workflow Service Creation' do
+  let(:druid) { 'qq613vj0238' } # a fixture Dor::Item record
+  let(:pid) { DruidTools::Druid.new(druid).druid }
+  # This spec doesn't work with a mock Dor::Item
+  # let(:item) { instantiate_fixture(druid, Dor::Item) }
+  # allow(Dor).to receive(:find).with(pid).and_return(item)
+
+  before :each do
+    admin_user # see spec_helper
   end
-  scenario 'redirect and display on show page' do
-    visit add_workflow_item_path 'druid:qq613vj0238'
-    click_button 'Add'
+
+  scenario 'redirect and display on show page - with JS', js: true do
+    visit add_workflow_item_path(pid)
+    expect(page).to have_content('Add workflow')
+    expect(page).to have_button('Add')
+    find('#wf').find(:option, 'accessionWF').select_option
+    find('#add_wf_button').trigger('click')
     within '.flash_messages' do
       expect(page).to have_css '.alert.alert-info', text: 'Added accessionWF'
     end
```

Comment:
This spec was giving weird results with or without JS.  Through trial-n-error, the resulting spec passed and consistently passes now.  It was a revelation to realize that it did not require an item mock, because the fedora-test already contains the item.